File: api1.js
import React, { useState } from 'react';
import axios from 'axios';

function StressLevelPrediction() {
    // Define state variables for form inputs and prediction result
    const [humidity, setHumidity] = useState('');
    const [temperature, setTemperature] = useState('');
    const [stepCount, setStepCount] = useState('');
    const [prediction, setPrediction] = useState('');
    const [error, setError] = useState('');

    // Handle form submission
    const handleSubmit = async (e) => {
        e.preventDefault();  // Prevent the form from submitting the usual way

        // Create the data object to send to the API
        const requestData = {
            Humidity: humidity,
            Temperature: temperature,
            Step_count: stepCount
        };

        try {
            // Send a POST request to the Flask API
            const response = await axios.post('http://localhost:5000/predict', requestData);
            
            // Set the prediction state with the API response
            setPrediction(response.data.diagnosis);
            setError('');
        } catch (error) {
            // Handle any errors that occur during the API request
            console.error('Error fetching prediction:', error);
            setError('Failed to fetch prediction. Please try again.');
        }
    };

    return (
        <div>
            <h1>Stress Level Prediction</h1>

            {/* Form for user inputs */}
            <form onSubmit={handleSubmit}>
                <div>
                    <label>Humidity:</label>
                    <input
                        type="text"
                        value={humidity}
                        onChange={(e) => setHumidity(e.target.value)}
                        required
                    />
                </div>

                <div>
                    <label>Body Temperature:</label>
                    <input
                        type="text"
                        value={temperature}
                        onChange={(e) => setTemperature(e.target.value)}
                        required
                    />
                </div>

                <div>
                    <label>Number of Steps:</label>
                    <input
                        type="text"
                        value={stepCount}
                        onChange={(e) => setStepCount(e.target.value)}
                        required
                    />
                </div>

                {/* Submit button */}
                <button type="submit">Predict</button>
            </form>

            {/* Display the prediction result */}
            {prediction && (
                <div>
                    <h2>Prediction Result:</h2>
                    <p>{prediction}</p>
                </div>
            )}

            {/* Display any error messages */}
            {error && (
                <div style={{ color: 'red' }}>
                    <p>{error}</p>
                </div>
            )}
        </div>
    );
}

export default StressLevelPrediction;

File: api1.py
from flask import Flask, request, jsonify
import numpy as np
import pickle

# Load the pre-trained model
model_path = r'C:\Users\aryan\OneDrive\Desktop\Algorithm ACES\stress_trained.sav'
loaded_model = pickle.load(open(model_path, 'rb'))

# Initialize Flask app
app = Flask(__name__)

# Define stress level prediction function
def stresslevel_prediction(input_data):
    # Convert input data to a numpy array and reshape it
    id_np_array = np.asarray(input_data)
    id_reshaped = id_np_array.reshape(1, -1)
    
    # Make prediction using the loaded model
    prediction = loaded_model.predict(id_reshaped)
    
    # Return prediction result
    if prediction[0] == 0:
        return "Stress Level: LOW"
    elif prediction[0] == 1:
        return "Stress Level: MEDIUM"
    else:
        return "Stress Level: HIGH"

# Define /predict endpoint
@app.route('/predict', methods=['POST'])
def predict():
    # Get JSON data from the request body
    data = request.get_json()

    # Extract input values from the JSON data
    Humidity = data.get('Humidity')
    Temperature = data.get('Temperature')
    Step_count = data.get('Step_count')
    
    # Convert input values to a list
    input_data = [Humidity, Temperature, Step_count]
    
    # Get stress level prediction
    diagnosis = stresslevel_prediction(input_data)
    
    # Return prediction result as a JSON response
    return jsonify({"diagnosis": diagnosis})

# Run the Flask app
if __name__ == '__main__':
    app.run(debug=True)
